### PR TITLE
audit: check for closed issues in comments

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----
This should help finding outdated patches in Homebrew/core. I couldn't get the API to work using the `GitHub::API` class so posting here for help with that.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
I feel like this should be a `brew audit` instead of RuboCop, unfortunately. It feels very weird to shell out to `curl` as part of RuboCop.

**Comment by SMillerDev** [comment]
Can we audit comments though?

**Comment by carlocab** [comment]
This audit should also be gated behind `--online`.

File: Library/Homebrew/formula_auditor.rb
Comment (SMillerDev) [inline Library/Homebrew/formula_auditor.rb:829]:
```suggestion
      return if repo == formula.tap.repo
```

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
@SMillerDev Looks good! Good to merge when you are.

File: Library/Homebrew/formula_auditor.rb
Comment (SMillerDev) [reply Library/Homebrew/formula_auditor.rb:834]: I think that really depends what we're commenting. If it's a workaround: "use the fix upstream deemed good enough to close the issue", if it's a dependency: "re-evaluate the dependency". 

I mostly see this being useful as a warning for patches that are merged but still apply.

Comment (MikeMcQuaid) [reply Library/Homebrew/formula_auditor.rb:834]: > I mostly see this being useful as a warning for patches that are merged but still apply.

Is it worth scoping just to PRs?

Might be worth some sort of nudge on the action but not sure how to make it generic enough?

All this said: fine to merge as-is and iterate.

Comment (SMillerDev) [reply Library/Homebrew/formula_auditor.rb:834]: I'm hoping one of the people that do words good will help once this is merged 😄.

> Is it worth scoping just to PRs?

If you mean upstream PRs, I don't think it would be a good trade-off to get better wording instead of a broader scope to catch this issue.

> All this said: fine to merge as-is and iterate.

Planning to write some small test and ensure that it really only catches comments and then I'll merge.

**Comment by SMillerDev** [comment]
This also fixes the audit above since it relied on `to_s` returning the formula content

**Comment by SMillerDev** [comment]
@MikeMcQuaid could you re-run this?

**Comment by MikeMcQuaid** [comment]
@SMillerDev reran

**Comment by SMillerDev** [comment]
Whoop whoop, it works. I'd like to appologize in advance to my Homebrew/core colleagues, I think this will unveil a bunch of forgotten patches/hacks.

**Comment by Bo98** [comment]
Oh misread an error, sorry about that.

**Comment by brendongo** [comment]
Should we also exclude links to other repos that are also discussions? Maybe special casing if the word before the url is `Relevant:`?

**Comment by nandahkrishna** [comment]
Adding to `@brendongo`'s comment, cases like [this URL](https://github.com/Homebrew/homebrew-core/pull/122602/commits/5a0f180dd072ce0bb879ac2d8f421586eb280d6c) which direct contributors to some discussion in an external repository may be useful to retain in formulae. Perhaps we could have a way to allow some of these URLs as suggested in the previous comment?

**Comment by carlocab** [comment]
Yea, it seems useful to be able to reference external issues as context for why we do things in formulae.

Alternatively these external issues can be linked in a comment in the PR, and the PR can be linked in the formula, but that seems needlessly complicated.

**Comment by SMillerDev** [comment]
Yeah I'm fine adding special cases for references (although I'm not behind a pc for a couple of days) the main point was to alert us of outdated comments but sometimes those are still needed.

**Pull request merged**
